general/react_dnd.md

```diff
@@ -0,0 +1,170 @@
+
+    `item.type` is also required, only drop targets registered for the same type will react to this drag.
+
+- `collect`: Optional. This is a function that receives `monitor` and `props`, and must return a plain object that you'll receive as the fist array element of the hook return value.
+
+### Connect drag with DOM element
+
+You can connect drag with an element in two differente ways:
+
+- Setting `drag` returned parameter as your element's `ref`.
+
+```
+return (
+    <div ref={drag}>
+        ...
+    </div>
+);
+```
+
+- Creating your own `ref`, calling `drag` returned parameter passing it out this ref and setting your element's ref with it.
+
+```
+const ref = useRef();
+
+drag(ref);
+
+return (
+    <div ref={ref}>
+        ...
+    </div>
+);
+```
+
+## useDrop
+
+Once the drag element is configured you can set your drop element(s) using `useDrop`.
+
+```
+const [{isOver}, drop] = useDrop({
+		accept: ACCEPTING_TYPES.ITEM,
+		canDrop(source, monitor) {
+			return isValidTarget(source, monitor);
+		},
+		collect: monitor => ({
+			isOver: !!monitor.isOver(),
+		}),
+		drop(source, monitor) {
+			if (monitor.canDrop()) {
+				onItemDrop(source, monitor);
+			}
+		},
+		hover(source, monitor) {
+			if (monitor.isOver() && monitor.canDrop()) {
+                onItemDropHover(source, monitor)
+			}
+		}
+	});
+```
+
+### Return value array
+
+`useDrop` returns an array of two values:
+
+- `Index 0`: An object containing collected properties from the collect function. If no collect function is defined, an empty object is returned.
+
+- `Index 1`: A connector function for the drop target. This **must** be attached to the drop-target portion of the DOM.
+
+### Hook config object (most important properties)
+
+- `accept`: Required. This must be set with the item type you want drop to react to.
+
+- `canDrop`: Optional. Use it to specify whether the drop target is able to accept the item. If you want to always allow it, just omit this method.
+
+- `collect`: Optional. This is a function that receives `monitor` and `props`, and must return a plain object that you'll receive as the fist array element of the hook return value.
+
+- `drop`: Called when a compatible item is dropped on the target.
+
+- `hover`: Called when an item is hovered over the component. You can check `monitor.isOver({ shallow: true })` to test whether the hover happens over just the current target, or over a nested one. Unlike drop(), this method will be called even if `canDrop()` is defined and returns false. You can check `monitor.canDrop()` to test whether this is the case.
+
+### Connect drop with DOM element
+
+You can connect drop with an element in two differente ways:
+
+- Setting `drop` returned parameter as your element's `ref`.
+
+```
+return (
+    <div ref={drop}>
+        ...
+    </div>
+);
+```
+
+- Creating your own `ref`, calling `drop` returned parameter passing it out this ref and setting your element's ref with it.
```

Comment:
Same issue as above in the "Connect drag with DOM element" section. Whatever sentence you go with up there, you should do the same here.